sw/device/tests/chip_power_sleep_load.c

```diff
@@ -0,0 +1,377 @@
+
+  LOG_INFO("SW_INFO: RV Timer active");
+
+  // Alert Ping
+
+  dif_alert_handler_alert_t alerts[ALERT_HANDLER_PARAM_N_ALERTS];
+  dif_alert_handler_class_t alert_classes[ALERT_HANDLER_PARAM_N_ALERTS];
+  for (int i = 0; i < ALERT_HANDLER_PARAM_N_ALERTS; ++i) {
+    alerts[i] = i;
+    alert_classes[i] = kDifAlertHandlerClassA;
+  }
+
+  dif_alert_handler_local_alert_t loc_alerts[] = {
+      kDifAlertHandlerLocalAlertAlertPingFail};
+  dif_alert_handler_class_t loc_alert_classes[] = {kDifAlertHandlerClassB};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = 2000}};
+  // Escalation protocol (struct) for an alert class
+  dif_alert_handler_class_config_t class_config = {
+      .auto_lock_accumulation_counter = kDifToggleDisabled,
+      .accumulator_threshold = UINT16_MAX,
+      .irq_deadline_cycles = UINT32_MAX,
+      .escalation_phases = esc_phases,
+      .escalation_phases_len = ARRAYSIZE(esc_phases),
+      .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase1,
+  };
+  dif_alert_handler_class_config_t class_configs[] = {class_config,
+                                                      class_config};
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA,
+                                         kDifAlertHandlerClassB};
+
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = ARRAYSIZE(alerts),
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = UINT16_MAX,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+
+  bool is_locked;
+  CHECK_DIF_OK(
+      dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+  CHECK(is_locked, "Expected alerts to be locked");
+
+  LOG_INFO("SW_INFO: Alert ping is active");
+
+  // PWM
+
+  // Configuration struct for PWM general
+  static const dif_pwm_config_t config_ = {
+      .clock_divisor = 0,
+      .beats_per_pulse_cycle = 32,
+  };
+
+  // Configuration struct for a specific PWM channel
+  static const dif_pwm_channel_config_t default_ch_cfg_ = {
+      .duty_cycle_a = 0,
+      .duty_cycle_b = 0,
+      .phase_delay = 0,
+      .mode = kDifPwmModeFirmware,
+      .polarity = kDifPwmPolarityActiveHigh,
+      .blink_parameter_x = 0,
+      .blink_parameter_y = 0,
+  };
+  static const dif_pwm_channel_t kPwmChannel[PWM_PARAM_N_OUTPUTS] = {
+      kDifPwmChannel0, kDifPwmChannel1, kDifPwmChannel2,
+      kDifPwmChannel3, kDifPwmChannel4, kDifPwmChannel5,
+  };
+  // Duty cycle (arbitrary) values (in the beats)
+  static volatile const uint16_t kPwmDutycycle[PWM_PARAM_N_OUTPUTS] = {
```

Comment:
why is this volatile? it doesn't look like its overwritten in the testbench is it?

Reply:
done